src/main/java/com/github/jcustenborder/kafka/connect/spooldir/AbstractCleanUpPolicy.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright © 2016 Jeremy Custenborder (jcustenborder@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.jcustenborder.kafka.connect.spooldir;
+
+import com.google.common.io.Files;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+
+abstract class AbstractCleanUpPolicy implements Closeable {
+  private static final Logger log = LoggerFactory.getLogger(AbstractCleanUpPolicy.class);
+  protected final File inputFile;
+  protected final File errorPath;
+  protected final File finishedPath;
+
+
+  protected AbstractCleanUpPolicy(File inputFile, File errorPath, File finishedPath) {
+    this.inputFile = inputFile;
+    this.errorPath = errorPath;
+    this.finishedPath = finishedPath;
+  }
+
+
+  public static AbstractCleanUpPolicy create(AbstractSourceConnectorConfig config, InputFile inputFile) throws IOException {
+    final AbstractCleanUpPolicy result;
+    switch (config.cleanupPolicy) {
+      case MOVE:
+        result = new Move(inputFile.inputFile, config.errorPath, config.finishedPath);
+        break;
+      case DELETE:
+        result = new Delete(inputFile.inputFile, config.errorPath, config.finishedPath);
+        break;
+      case NONE:
+        result = new AbstractCleanUpPolicy.None(inputFile.inputFile, config.errorPath, config.finishedPath);
```

Comment:
Can this be `result = new None(...`, like the Move and Delete?

Reply:
Thanks! Corrected.